PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.webapp/src/com/microsoft/azuretools/webapp/ui/WebAppDeployDialog.java

```diff
@@ -534,9 +534,7 @@ protected IStatus run(IProgressMonitor monitor) {
                 monitor.beginTask(message, IProgressMonitor.UNKNOWN);
                 try {
                     AzureDeploymentProgressNotification.notifyProgress(this, deploymentName, sitePath, 5, message);
-                    if (MavenUtils.isMavenProject(project)) {
-                        MavenUtils.executePackageGoal(project);
-                    } else {
+                    if (!MavenUtils.isMavenProject(project)) {
```

Comment:
why not put this part into the handler? which makes consistency that the package is ready before we show the dialog?